spinner/spinner.go

```diff
@@ -23,12 +24,15 @@ import (
 // ⣾  Loading...
 type Spinner struct {
 	spinner    spinner.Model
-	action     func()
+	action     func(ctx context.Context, w io.Writer) error
```

Comment:
I'm unsure about this

Some may need a spinner, most won't. Context is there to be able to cancel things

Maybe, if you want to do return string, don't add a buffer in the signature and in the spinner, but in the returned value

So something like

```suggestion
	action     func(ctx context.Context) (string, error)
```

Or refactor everything to pass the spinner itself as argument

Reply:
its not to return the string, its to print above the spinner